backend/pkg/contenttype/middleware.go

```diff
@@ -0,0 +1,49 @@
+package contenttype
+
+import (
+	"errors"
+	"mime"
+	"net/http"
+	"slices"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/mendersoftware/mender-server/pkg/rest.utils"
+)
+
+func CheckJSON() gin.HandlerFunc {
+	return Middleware(false, "application/json")
+}
+
+// For stronger checking with charset, checkCharest should be true.
+// The charset should be uppercase.
+func Middleware(checkCharest bool, contentTypes ...string) gin.HandlerFunc {
```

Comment:
Small typo :upside_down_face:
```suggestion
func Middleware(checkCharset bool, contentTypes ...string) gin.HandlerFunc {
```